### PR TITLE
feat: Implement basics of pre-commit

### PR DESCRIPTION
**Pull request by dcramer**
- Add pre-commit framework
- Implement a number of reasonable defaults and exclusions
- Move CI generated artifacts into .artifacts/
- Run pre-commit over entirety of codebase (and apply various fixes)
- Remove dead runtests.js

### CODE REVIEW

**Comment by dcramer** [comment]
Putting this up for feedback. It's only a step towards replacing our linters with pre-commit.

- eslint/prettier could probably be swapped out for the hooks I built (which are used by Zeus)
- flake8 needs proper registration of our style checks in order to swap out (and likely those style checks need moved into another repo)
- black is included as an example, but we're not using it yet (we've talked about wanting to move to it)

See also https://github.com/pre-commit/pre-commit-hooks

File: .coveragerc [hunk 1]
```diff
@@ -1,2 +0,0 @@
-[report]
```
Comment (dcramer) [inline .coveragerc]: moved into setup.cfg

File: .travis.yml [hunk 2]
```diff
@@ -51,14 +51,13 @@ after_failure:
...
 after_script:
-  - zeus upload -t "text/xml+xunit" junit.xml
-    -t "text/xml+xunit" jest.junit.xml
-    -t "text/xml+coverage" coverage.xml
-    -t "text/xml+coverage" coverage/cobertura-coverage.xml
-    -t "text/html+pytest" pytest.html
-    -t "text/plain+pycodestyle" flake8.pycodestyle.log
-    -t "text/xml+checkstyle" eslint.checkstyle.xml
-    -t "application/webpack-stats+json" webpack-stats.json
+  - zeus upload -t "text/xml+xunit" .artifacts/*.junit.xml
```
Comment (dcramer) [inline .travis.yml]: need to confirm zeus isnt missing artifacts now

File: package.json [hunk 3]
```diff
@@ -142,6 +142,9 @@
       "<rootDir>/node_modules/reflux"
     ]
   },
+  "jest-junit": {
+    "output": ".artifacts/jest.junit.xml"
```
Comment (dcramer) [inline package.json]: i couldnt actually figure out why its 'jest.junit.xml' without this, but its possible we're running a version older than what their documentation shows

File: src/sentry/static/sentry/debug/icons/demo.js [hunk 4]
```diff
@@ -1,30 +1,29 @@
 if (!('boxShadow' in document.body.style)) {
```
Comment (dcramer) [inline src/sentry/static/sentry/debug/icons/demo.js]: very possible this file isnt even used?

File: .artifacts/README [hunk 5]
```diff
@@ -0,0 +1 @@
+This folder captures artifacts generated from .pre-commit, such as lint violations. It is primarily used for continuous integration services.
```
Comment (mattrobenolt) [inline .artifacts/README]: Don't we wanna gitignore this directory?

Comment (mattrobenolt) [reply .artifacts/README]: Oh, you did below, but this is still staged with it.

**Review by mattrobenolt — APPROVED** [review]
:ok:

**Comment by dcramer** [comment]
For documentation...

I intend to (given time) follow up and:

- Move sentry's python lint hooks into their own library, and making that runnable within pre-commit
- Move our yarn.lock (and any other) check into our pre-commit-hooks repo
- Enable JS/Python via pre-commit
- Sort out the Black conversion (assuming we're still bought in), or alternatively run autopep8 via pre-commit

**Pull request merged**